// @material-ui/unstyled/node/BadgeUnstyled/badgeUnstyledClasses.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getBadgeUtilityClass = getBadgeUtilityClass;
exports.default = void 0;

function getBadgeUtilityClass(name) {
  return `MuiBadge-${name}`;
}

const badgeUnstyledClasses = {
  root: getBadgeUtilityClass('root'),
  badge: getBadgeUtilityClass('badge'),
  dot: getBadgeUtilityClass('dot'),
  anchorOriginTopLeftCircular: getBadgeUtilityClass('anchorOriginTopLeftCircular'),
  anchorOriginTopLeftRectangular: getBadgeUtilityClass('anchorOriginTopLeftRectangular'),
  anchorOriginTopRightCircular: getBadgeUtilityClass('anchorOriginTopRightCircular'),
  anchorOriginTopRightRectangular: getBadgeUtilityClass('anchorOriginTopRightRectangular'),
  anchorOriginBottomLeftCircular: getBadgeUtilityClass('anchorOriginBottomLeftCircular'),
  anchorOriginBottomLeftRectangular: getBadgeUtilityClass('anchorOriginBottomLeftRectangular'),
  anchorOriginBottomRightCircular: getBadgeUtilityClass('anchorOriginBottomRightCircular'),
  anchorOriginBottomRightRectangular: getBadgeUtilityClass('anchorOriginBottomRightRectangular'),
  invisible: getBadgeUtilityClass('invisible')
};
var _default = badgeUnstyledClasses;
exports.default = _default;
