// @material-ui/unstyled/node/BadgeUnstyled/BadgeUnstyled.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var React = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _clsx = _interopRequireDefault(require("clsx"));

var _utils = require("@material-ui/utils");

var _isHostComponent = _interopRequireDefault(require("../utils/isHostComponent"));

var _badgeUnstyledClasses = _interopRequireWildcard(require("./badgeUnstyledClasses"));

const useBadgeClasses = props => {
  const {
    variant,
    anchorOrigin,
    overlap,
    invisible,
    classes = {}
  } = props;
  const utilityClasses = {
    root: (0, _clsx.default)(_badgeUnstyledClasses.default['root'], classes['root']),
    badge: (0, _clsx.default)(_badgeUnstyledClasses.default['badge'], classes['badge'], (0, _badgeUnstyledClasses.getBadgeUtilityClass)(variant), _badgeUnstyledClasses.default[`anchorOrigin${(0, _utils.unstable_capitalize)(anchorOrigin.vertical)}${(0, _utils.unstable_capitalize)(anchorOrigin.horizontal)}${(0, _utils.unstable_capitalize)(overlap)}`], classes[`anchorOrigin${(0, _utils.unstable_capitalize)(anchorOrigin.vertical)}${(0, _utils.unstable_capitalize)(anchorOrigin.horizontal)}${(0, _utils.unstable_capitalize)(overlap)}`], invisible && [_badgeUnstyledClasses.default['invisible'], classes.invisible])
  };
  return utilityClasses;
};

const BadgeUnstyled = /*#__PURE__*/React.forwardRef(function BadgeUnstyled(props, ref) {
  const {
    anchorOrigin: anchorOriginProp = {
      vertical: 'top',
      horizontal: 'right'
    },
    classes: classesProp = {},
    badgeContent: badgeContentProp,
    component: Component = 'span',
    children,
    className,
    components = {},
    componentsProps = {},
    invisible: invisibleProp,
    max: maxProp = 99,
    overlap: overlapProp = 'rectangular',
    showZero = false,
    variant: variantProp = 'standard',

    /* eslint-disable react/prop-types */
    theme
  } = props,
        other = (0, _objectWithoutPropertiesLoose2.default)(props, ["anchorOrigin", "classes", "badgeContent", "component", "children", "className", "components", "componentsProps", "invisible", "max", "overlap", "showZero", "variant", "theme"]);
  const prevProps = (0, _utils.usePreviousProps)({
    anchorOrigin: anchorOriginProp,
    badgeContent: badgeContentProp,
    max: maxProp,
    overlap: overlapProp,
    variant: variantProp
  });
  let invisible = invisibleProp;

  if (invisibleProp == null && (badgeContentProp === 0 && !showZero || badgeContentProp == null && variantProp !== 'dot')) {
    invisible = true;
  }

  const {
    anchorOrigin = anchorOriginProp,
    badgeContent,
    max = maxProp,
    overlap = overlapProp,
    variant = variantProp
  } = invisible ? prevProps : props;
  const stateAndProps = (0, _extends2.default)({}, props, {
    anchorOrigin,
    badgeContent,
    invisible,
    max,
    overlap,
    variant
  });
  let displayValue = '';

  if (variant !== 'dot') {
    displayValue = badgeContent > max ? `${max}+` : badgeContent;
  }

  const classes = useBadgeClasses((0, _extends2.default)({}, stateAndProps, {
    classes: classesProp
  }));
  const Root = components.Root || Component;
  const rootProps = componentsProps.root || {};
  const Badge = components.Badge || 'span';
  const badgeProps = componentsProps.badge || {};
  return /*#__PURE__*/React.createElement(Root, (0, _extends2.default)({}, rootProps, !(0, _isHostComponent.default)(Root) && {
    as: Component,
    styleProps: (0, _extends2.default)({}, stateAndProps, rootProps.styleProps),
    theme
  }, {
    ref: ref
  }, other, {
    className: (0, _clsx.default)(classes.root, rootProps.className, className)
  }), children, /*#__PURE__*/React.createElement(Badge, (0, _extends2.default)({}, badgeProps, !(0, _isHostComponent.default)(Badge) && {
    styleProps: (0, _extends2.default)({}, stateAndProps, badgeProps.styleProps),
    theme
  }, {
    className: (0, _clsx.default)(classes.badge, badgeProps.className)
  }), displayValue));
});
process.env.NODE_ENV !== "production" ? BadgeUnstyled.propTypes = {
  // ----------------------------- Warning --------------------------------
  // | These PropTypes are generated from the TypeScript type definitions |
  // |     To update them edit the d.ts file and run "yarn proptypes"     |
  // ----------------------------------------------------------------------

  /**
   * The anchor of the badge.
   * @default {
   *   vertical: 'top',
   *   horizontal: 'right',
   * }
   */
  anchorOrigin: _propTypes.default.shape({
    horizontal: _propTypes.default.oneOf(['left', 'right']).isRequired,
    vertical: _propTypes.default.oneOf(['bottom', 'top']).isRequired
  }),

  /**
   * The content rendered within the badge.
   */
  badgeContent: _propTypes.default.node,

  /**
   * The badge will be added relative to this node.
   */
  children: _propTypes.default.node,

  /**
   * Override or extend the styles applied to the component.
   * @default {}
   */
  classes: _propTypes.default.object,

  /**
   * @ignore
   */
  className: _propTypes.default.string,

  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: _propTypes.default.elementType,

  /**
   * The components used for each slot inside the Badge.
   * Either a string to use a HTML element or a component.
   * @default {}
   */
  components: _propTypes.default.shape({
    Badge: _propTypes.default.elementType,
    Root: _propTypes.default.elementType
  }),

  /**
   * The props used for each slot inside the Badge.
   * @default {}
   */
  componentsProps: _propTypes.default.object,

  /**
   * If `true`, the badge is invisible.
   */
  invisible: _propTypes.default.bool,

  /**
   * Max count to show.
   * @default 99
   */
  max: _propTypes.default.number,

  /**
   * Wrapped shape the badge should overlap.
   * @default 'rectangular'
   */
  overlap: _propTypes.default.oneOf(['circular', 'rectangular']),

  /**
   * Controls whether the badge is hidden when `badgeContent` is zero.
   * @default false
   */
  showZero: _propTypes.default.bool,

  /**
   * The variant to use.
   * @default 'standard'
   */
  variant: _propTypes.default
  /* @typescript-to-proptypes-ignore */
  .oneOfType([_propTypes.default.oneOf(['dot', 'standard']), _propTypes.default.string])
} : void 0;
var _default = BadgeUnstyled;
exports.default = _default;
